pkg/kclient/volumes_test.go

```diff
@@ -0,0 +1,330 @@
+
+			err := AddVolumeMountToPodTemplateSpec(podTemplateSpec, tt.volumeName, tt.pvc, tt.containerMountPathsMap)
+			if !tt.wantErr && err != nil {
+				t.Errorf("TestAddVolumeMountToPodTemplateSpec.AddVolumeMountToPodTemplateSpec() unexpected error %v, wantErr %v", err, tt.wantErr)
+			}
+
+			mountPathCount := 0
+			for _, podTempSpecContainer := range podTemplateSpec.Spec.Containers {
+				if podTempSpecContainer.Name == tt.container.Name {
+					for _, volumeMount := range podTempSpecContainer.VolumeMounts {
+						if volumeMount.Name == tt.volumeName {
+							for _, mountPath := range tt.containerMountPathsMap[tt.container.Name] {
+								if volumeMount.MountPath == mountPath {
+									mountPathCount++
+								}
+							}
+						}
+					}
+				}
+			}
+
+			if mountPathCount != len(tt.containerMountPathsMap[tt.container.Name]) {
+				t.Errorf("Volume Mounts for %s have not been properly mounted to the podTemplateSpec", tt.volumeName)
+			}
+		})
+	}
+}
+
+func TestGetPVCsFromSelector(t *testing.T) {
+	tests := []struct {
+		name      string
+		pvcName   string
+		size      string
+		namespace string
+		labels    map[string]string
+		wantErr   bool
+	}{
+		{
+			name:      "Case: Valid pvc name",
+			pvcName:   "mypvc",
+			size:      "1Gi",
+			namespace: "default",
+			labels: map[string]string{
+				"mylabel1": "testpvc1",
+				"mylabel2": "testpvc2",
+			},
+			wantErr: false,
+		},
+		{
+			name:      "Case: Wrong Label Selector",
+			pvcName:   "mypvc",
+			size:      "1Gi",
+			namespace: "default",
+			labels: map[string]string{
+				"mylabel1": "testpvc1",
+				"mylabel2": "testpvc2",
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// initialising the fakeclient
+			fkclient, fkclientset := FakeNew()
+			fkclient.Namespace = tt.namespace
+
+			selector := util.ConvertLabelsToSelector(tt.labels)
+
+			listOfPVC := corev1.PersistentVolumeClaimList{
+				Items: []corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:   tt.pvcName,
+							Labels: tt.labels,
+						},
+					},
+				},
+			}
+
+			fkclientset.Kubernetes.PrependReactor("list", "persistentvolumeclaims", func(action ktesting.Action) (bool, runtime.Object, error) {
+				if tt.name == "Case: Wrong Label Selector" {
+					return true, nil, fmt.Errorf("TestGetPVCsFromSelector: Labels do not match with expected values, expected:%s, got:%s", selector, selector+",garbage=true")
+				}
+				return true, &listOfPVC, nil
+			})
+
+			PVCs, err := fkclient.GetPVCsFromSelector(selector)
+			if !tt.wantErr && err != nil {
+				t.Errorf("TestGetPVCsFromSelector: Error listing PVCs with selector: %v", err)
+			}
+
+			if len(PVCs) == 0 || len(PVCs) > 1 {
+				if !tt.wantErr {
+					t.Errorf("TestGetPVCsFromSelector: Incorrect amount of PVC found with selector %s", selector)
+				} else {
```

Comment:
is this else needed? it seems to just be returning? which would happen anyway if the `if statement` is not executed.

Reply:
@girishramnani I've updated the commit, thx!